app/forms/state_file/az_retirement_income_form.rb

```diff
@@ -0,0 +1,33 @@
+module StateFile
+  class AzRetirementIncomeForm < QuestionsForm
+    set_attributes_for :intake,
+                       :received_military_retirement_payment,
+                       :received_military_retirement_payment_amount,
+                       :primary_received_pension,
+                       :primary_received_pension_amount,
+                       :spouse_received_pension,
+                       :spouse_received_pension_amount
+
+    validates :received_military_retirement_payment_amount, presence: true, numericality: true, allow_blank: false, if: -> { received_military_retirement_payment == "yes" }
+    validates :primary_received_pension_amount, presence: true, numericality: true, allow_blank: false, if: -> { primary_received_pension == "yes" }
+    validates :spouse_received_pension_amount, presence: true, numericality: true, allow_blank: false, if: -> { spouse_received_pension == "yes" }
+    validate :below_1040_amount, if: -> { [received_military_retirement_payment_amount, primary_received_pension_amount, spouse_received_pension_amount].any?(&:present?) }
```

Comment:
NIT: Consider having a `total_pension_amount` reader which does

```ruby
def total_pension_amount
  received_military_retirement_payment_amount&.to_i + primary_received_pension_amount&.to_i + spouse_received_pension_amount&.to_i
end
```

and then check `total_pension_amount > 0`. This will make the validation a little easier to read and enables a tidier `if` check on line 25.

Reply:
question about that! part of having the `below_1040_amount` method is that it attaches 4 different errors: 3 blank ones so that the red line appears next to the offending inputs, 1 with a message that shows at the bottom because it's about the sum of all 3. how would i assign those errors to the right attributes with a one-line validation?

Reply:
actually i've just realized i don't need to have this `if` check on line 14 because if the amounts are nil they'll be converted to 0 and the sum when all are absent will always be less than fed_taxable_pensions (because this page only shows if fed_taxable_pensions > 0)